packages/replayio/src/utils/browser/launchBrowser.ts

```diff
@@ -42,42 +45,67 @@ export async function launchBrowser(
     throw new Error(`Replay browser not found at: ${browserExecutablePath}`);
   }
 
-  debug(
-    `Launching browser: ${browserExecutablePath} with args:\n`,
-    args.join("\n"),
-    "\n",
-    processOptions
-  );
+  const processes = await findProcess("name", browserExecutablePath);
+  if (processes.length > 0) {
+    const match = processes[0];
 
-  // Wait until the user quits the browser process OR
-  // until the user presses a key to continue (in which case, we will kill the process)
-  const abortControllerForPrompt = new AbortController();
+    debug(`Browser process already running at ${highlight(match.pid)}`);
 
-  const spawnDeferred = spawnProcess(browserExecutablePath, args, processOptions, {
-    onSpawn: () => {
-      if (process.stdin.isTTY) {
-        console.log(`Recording... ${dim("(press any key to stop recording)")}`);
+    console.log(`Recording... ${dim("(quit the Replay Browser to stop recording)")}`);
 
-        prompt({
-          signal: abortControllerForPrompt.signal,
-        }).then(() => {
-          spawnDeferred.data.kill();
-        });
-      } else {
-        console.log(`Recording... ${dim("(quit the Replay Browser to stop recording)")}`);
-      }
-    },
-    printStderr: (text: string) => {
-      debug(stderrPrefix("stderr"), text);
-    },
-    printStdout: (text: string) => {
-      debug(stdoutPrefix("stdout"), text);
-    },
-  });
+    // Ask the browser to open a new tab
+    spawnProcess(browserExecutablePath, args, processOptions);
+
+    // The best we can do in this case is to regularly poll to see when the process exits
+    await retryWithLinearBackoff(
+      async () => {
+        await timeoutAfter(1_000);
+        const processes = await findProcess("name", browserExecutablePath);
+        if (processes.length > 0) {
+          throw Error("Still running");
+        }
+      },
+      undefined,
+      Number.POSITIVE_INFINITY
+    );
```

Comment:
nit: I'd probably prefer a simple loop here instead of throwing an error to trigger the retry. I feel like with a simple loop there would be less moving pieces and ultimately it would be simpler and easier to understand.